end_to_end_tests/golden-record/my_test_api_client/models/a_model.py

```diff
@@ -217,17 +225,28 @@ def _parse_one_of_models(data: object) -> Union[FreeFormModel, ModelWithUnionPro
                 return one_of_models_type_0
             except:  # noqa: E722
                 pass
-            if not isinstance(data, dict):
+            try:
+                one_of_models_type_1: ModelWithUnionProperty
+                if not isinstance(data, dict):
+                    raise TypeError()
+                one_of_models_type_1 = ModelWithUnionProperty.from_dict(data)
+
+                return one_of_models_type_1
+            except:  # noqa: E722
+                pass
+            if not True:
```

Comment:
I know this generated is a little silly, but 🤷 it reduces the complexity of the JINJA templating